scraper/doctolib/doctolib.py

```diff
@@ -94,26 +93,25 @@ def fetch(self, request: ScraperRequest) -> Optional[str]:
 
         if visit_motive_ids is None:
             return None
-        # practice_ids / agenda_ids
-        agenda_ids, practice_ids = _find_agenda_and_practice_ids(
-            data, visit_motive_ids, practice_id_filter=practice_id
-        )
-        if not agenda_ids or not practice_ids:
-            return None
+
         all_agendas = parse_agenda_ids(rdata)
-        agenda_ids = self.sort_agenda_ids(all_agendas, agenda_ids)
 
-        # temporary_booking_disabled ??
+        first_availability = None
+        for visit_motive_id in visit_motive_ids:
+            agenda_ids, practice_ids = _find_agenda_and_practice_ids(
+                data, visit_motive_id, practice_id_filter=practice_id
+            )
+            agenda_ids = self.sort_agenda_ids(all_agendas, agenda_ids)
 
-        agenda_ids_q = "-".join(agenda_ids)
-        practice_ids_q = "-".join(practice_ids)
-        start_date = request.get_start_date()
+            # temporary_booking_disabled ??
 
-        first_availability = None
-        start_date_tmp = start_date
-        for motive_id in visit_motive_ids:
+            agenda_ids_q = "-".join(agenda_ids)
```

Comment:
Tu prends que les derniers agenda_ids de la dernière exécution de la boucle `for visit_motive_id in visit_motive_ids:` ? ça ne prend pas en compte les agenda_ids reliés au `motive_id` ?

Reply:
Il me semble pas bien comprendre ta modification, tu n'assignes pas les agenda_id pour chaque motif ?

Car il est possible d'avoir plusieurs motifs comme par ex:

`
1ère injection AstraZeneca
2ème injection AstraZeneca
1ère injection Moderna
1ère injection Pfizer
2ème injection Pfizer
2ème injection Moderna
`

On doit vérifier les disponibilités sur les trois motives séparément (c'est déjà filtré), en prenant les agenda_ids reliés de ces trois motives :
`
1ère injection AstraZeneca
1ère injection Moderna
1ère injection Pfizer
`

A vérifier le cas des mauvais agenda_ids retournés, on a eu le même souci avec les pratice id

Reply:
C'est plus facile de lire directement le fichier que les différences là. Regarde [ici](https://github.com/CovidTrackerFr/vitemadose/blob/7aa597d36f19dee0eab50837ce6a5a445287e570/scraper/doctolib/doctolib.py#L100-L103). On prenait tous les practice ids et agendas ids tant qu'ils avaient un des motifs avant, ce qui faisait que des fois on faisait des query trop larges (je pense qu'ici, on avait un agenda qui devait faire du AZ et Pfizer avant, mais qui ne fait que du AZ (non dispo), et on le query encore pour le Pfizer, qui existe encore sur l'api alors que non dispo).

Donc en gros, ouais je fais exactement ce que tu penses que le code devrait faire :)